uiautomationutilities/src/main/java/com/microsoft/identity/client/ui/automation/app/OneAuthTestApp.java

```diff
@@ -0,0 +1,200 @@
+        try {
+            // Add a delay so that token can be retrived successfully
+            Thread.sleep(2000);
+        } catch (InterruptedException e) {
+            throw new RuntimeException(e);
+        }
+        return getTokeSecret();
+    }
+
+    private void signIn(@NonNull final String username,
+                        @NonNull final String password,
+                        @NonNull final FirstPartyAppPromptHandlerParameters promptHandlerParameters) {
+        Logger.i(TAG, "Adding username and password on login screen..");
+
+        try {
+            final UiObject emailField = UiAutomatorUtils.obtainUiObjectWithTextAndClassType(
+                    "", EditText.class);
+            emailField.setText(username);
+            final UiObject nextBtn = UiAutomatorUtils.obtainUiObjectWithTextAndClassType(
+                    "Next", Button.class);
+            nextBtn.click();
+        } catch (UiObjectNotFoundException e) {
+            throw new RuntimeException(e);
+        }
+
+        Logger.i(TAG, "Handle AAD Login page prompt..");
+        // handle prompt in AAD login page
+        new AadPromptHandler(promptHandlerParameters).handlePrompt(username, password);
+    }
+
+    public void handleBackButton() {
+        UiAutomatorUtils.pressBack();
+    }
+
+    @Override
+    public void addAnotherAccount(String username, String password, FirstPartyAppPromptHandlerParameters promptHandlerParameters) {
+
+    }
+
+    @Override
+    public void onAccountAdded() {
+
+    }
+
+    @Override
+    public void confirmAccount(@NonNull String username) {
+        // Make sure we are seeing the output text view
+        final UiObject resultUIObject = UiAutomatorUtils.obtainUiObjectWithResourceId("com.microsoft.oneauth.testapp:id/txtGeneralInfo");
+        try {
+            Assert.assertTrue(resultUIObject.getText().contains("Result: Success"));
+        } catch (UiObjectNotFoundException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    public void assertSuccess() {
+        final UiObject resultUIObject = UiAutomatorUtils.obtainUiObjectWithResourceId("com.microsoft.oneauth.testapp:id/txtGeneralInfo");
+        final UiObject accountIdObject = UiAutomatorUtils.obtainUiObjectWithResourceId("com.microsoft.oneauth.testapp:id/txtAccountId");
+        try {
+            resultUIObject.waitForExists(FIND_UI_ELEMENT_TIMEOUT);
+            accountIdObject.waitForExists(FIND_UI_ELEMENT_TIMEOUT);
+            Assert.assertTrue(resultUIObject.getText().contains("Result: Success"));
+            Assert.assertFalse(TextUtils.isEmpty(resultUIObject.getText()));
+            Assert.assertFalse(TextUtils.isEmpty(accountIdObject.getText()));
+        } catch (UiObjectNotFoundException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    /**
+     * Returns string list of all accounts available
+     */
+    public void getAllAccounts() {
```

Comment:
i thought this is supposed to be returning the accounts. is that correct?

Reply:
Correct. refactored the code to return the accounts list.